presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java

```diff
@@ -515,6 +518,38 @@ private static NullableValue parsePartitionValue(
         return partitionValue == null ? NullableValue.asNull(prestoType) : NullableValue.of(prestoType, partitionValue);
     }
 
+    // Strip the constraints on info columns like "$path", "$data_sequence_number" from the list.
+    public static <U> TupleDomain<IcebergColumnHandle> getNonInfoColumnConstraints(TupleDomain<U> allConstraints)
+    {
+        return allConstraints.transform(c -> isMetadataColumnId(((IcebergColumnHandle) c).getId()) ? null : (IcebergColumnHandle) c);
+    }
+
+    public static <U> TupleDomain<IcebergColumnHandle> getInfoColumnConstraints(TupleDomain<U> allConstraints)
```

Comment:
I think calling it "getMetadataColumnConstraints" is better. It's been called "MetadataColumn" in Presto forever, e.g. isMetadataColumnId(). Giving it a new name "InfoColumn" would confuse the readers a little bit. Will you be able to send  commit to unify the names in the HIve change as well?